Makefile

```diff
@@ -40,51 +40,127 @@ buildnocache:
 	./ci_cd/version.sh
 	@docker-compose build --no-cache --pull
 
+build:
+	./ci_cd/version.sh
+	@docker-compose build
+
 up:
 	docker-compose up -d
 
+logs:
+	@docker-compose logs -f $(API_SERVICE)
+
 start:
 	@make up
 
+restart_api:
+	@docker-compose restart api_service
+
+# -----------------
+# DB actions
+# -----------------
 dbbackup:
-	@docker-compose exec --user=$(CURRENT_UID) $(API_SERVICE) python manage.py dbbackup local
+	@docker-compose run \
+		--rm \
+		--name api_dbbackup \
+		-e AWS_ACCESS_KEY_ID=${OLD_AWS_ACCESS_KEY_ID} \
+		-e AWS_SECRET_ACCESS_KEY=${OLD_AWS_SECRET_ACCESS_KEY} \
+		-e PGPASSWORD=postgres \
+		--entrypoint python \
+		--user=$(CURRENT_UID) \
+		$(API_SERVICE) \
+		manage.py dbbackup local
 
 dbrestore:
-	@docker-compose exec --user=$(CURRENT_UID) $(API_SERVICE) python manage.py dbrestore local
+	@docker-compose run \
+		--rm \
+		--name api_dbrestore \
+		-e AWS_ACCESS_KEY_ID=${OLD_AWS_ACCESS_KEY_ID} \
+		-e AWS_SECRET_ACCESS_KEY=${OLD_AWS_SECRET_ACCESS_KEY} \
+		-e PGPASSWORD=postgres \
+		--entrypoint python \
+		--user=$(CURRENT_UID) \
+		$(API_SERVICE) \
+		manage.py dbrestore local
 
 migrate:
-	@docker-compose exec --user=$(CURRENT_UID) $(API_SERVICE) python manage.py migrate
-
+	@docker-compose run \
+		--rm \
+		--name api_migrate \
+		--entrypoint python \
+		--user=$(CURRENT_UID) \
+		$(API_SERVICE) \
+		manage.py migrate
+
+make_migrations:
+	@docker-compose run \
+		--rm \
+		--name api_make_migrations \
+		--entrypoint python \
+		--user=$(CURRENT_UID) \
+		$(API_SERVICE) \
+		manage.py makemigrations
+
+# -----------------
+# Setup actions
+# -----------------
 install:
 	@echo "\n--- Shutting down existing stack ---\n"
 	@make down
 	@echo "\n--- Building new docker image ---\n"
-	@make buildnocache
-	@make up
-	@echo "\n--- Spinning up new stack ---\n"
-	@sleep 20
-	@echo "\n--- Applying MERMAID database migrations ---\n"
+	@make build
+	@echo "\n--- Migrate ---\n"
 	@make migrate
-
+	@echo "\n--- Spinning up new stack ---\n"
+	@make up
```

Comment:
Migrations are done when the container starts in the docker entrypoint in `entry.sh`

Reply:
docker run spins up a new isolated container, and docker exec run's the command in an existing container. For example, now when you call `make runserver`, it will spin up a new container that will be started with runserver (rather than gunicorn). I've left the docker-compose api_service to simulate what a deployed container looks like.